src/examples/edsl_extension_register_result.ml

```diff
@@ -0,0 +1,394 @@
+      ~configuration:Biokepi.Tools.Mutect.Configuration.default
+    |> register "mutect-vcf"
+
+  let run ~normal ~tumor =
+    observe (fun () ->
+        example
+          ~normal:(Library.fastq_of_input normal)
+          ~tumor:(Library.fastq_of_input tumor)
+      )
+end
+(*M
+
+The value `example` gets the expected type:
+
+```ocaml
+val example:
+   normal:[ `Fastq ] list Registrable_bfx.repr ->
+   tumor:[ `Fastq ] list Registrable_bfx.repr ->
+   [ `Vcf ] Registrable_bfx.repr
+```
+
+### Compiler to Graphviz
+
+We “inherit” the implementation of the parent EDSL (through `include`), and
+then we implement the extension.
+
+We decide to hide the `register` step from the displayed graph.
+M*)
+module To_dot_with_register = struct
+  include Biokepi.EDSL.Compile.To_dot
+  let register _ x = x
+end
+(*M
+### Compiler To Json
+
+This interpreter is slightly more complex; in order to save it the database, we
+want to save the intermediate JSON structure for later insertion in the
+database.
+
+We functorize the module to provide such an infrastructure (assuming the
+`Mem.add` registers the tuple `(metadata, json)` as a side-effect).
+
+Again we `include` the `To_json` compiler from Biokepi, and add the function
+`register`.
+M*)
+module To_json_with_register (Memory: sig 
+    val add : string -> Yojson.Basic.json -> unit
+  end) = struct
+  include Biokepi.EDSL.Compile.To_json
+  let register metadata x ~var_count =
+    let compiled = x ~var_count in
+    Memory.add metadata compiled;
+    compiled
+end
+
+(*M
+### Compiler To Ketrew Workflow
+
+Here is where the magic happens. As above, we `include` the compiler from the
+library to get the previously defined implementations.
+
+This time we expect an interface providing `Mem.look_up` to find a previously
+saved JSON datastructure.
+
+The function `register` creates an intermediary Ketrew workflow-node.
+
+- It provides the same product as the argument of the `register` function
+  (child node).
+- It has the parameter ``~equivalence:`None`` to make sure Ketrew does not
+  merge it with the child node.
+- We force the condition of the node to be “never already done:”
+  ``~done_when:`Never``.
+- It has two dependencies: the child node and the actual registration in the
+  database (from above `FDB.register`).ster`).
+
+
+M*)
+module To_workflow_with_register 
+    (Mem : sig val look_up: string -> Yojson.Basic.json end)
+    (Config : sig
+       include Biokepi.EDSL.Compile.To_workflow.Compiler_configuration
+       val results_db_uri : string
+     end)
+= struct
+  include Biokepi.EDSL.Compile.To_workflow.Make(Config)
+  open Biokepi.EDSL.Compile.To_workflow.File_type_specification
+  let register : type a . string -> a t -> a t = fun metadata x ->
+    let make_node any_node =
+      let registration =
+        let path = any_node#product#path in
+        let ketrew_id = Ketrew.EDSL.node_id any_node in
+        let json = Mem.look_up metadata in
```

Comment:
Might be good to have a note in the docs above this explaining that. How simple would it be for the register node to do the JSON compilation on the nodes it's interested in (so the preprocessing step isn't required)?